### PR TITLE
Add Programming Paradigms as concepts

### PR DESCRIPTION
**Pull request by senal**
Hi @iHiD,
Hope programming paradigms as concepts will add value to beginners to understand different programming styles and their benefits.

Let me know your feedback.

~Senal

### CODE REVIEW

File: reference/concepts/declarative_programming.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Declarative programming
```
Comment (ErikSchierboom) [inline reference/concepts/declarative_programming.md]: There is an existing document describing declarative programming: https://github.com/exercism/v3/blob/master/reference/paradigms/declarative.md Could you remove this document from the PR?

File: reference/concepts/imperative_programming.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Imperative programming
```
Comment (ErikSchierboom) [inline reference/concepts/imperative_programming.md]: There is an existing document describing declarative programming: https://github.com/exercism/v3/blob/master/reference/paradigms/imperative.md Could you remove this document from the PR?

File: languages/csharp/reference/README.md [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 The C# concept exercises are based on concepts. The list below contains the concepts that have been identified for the C# language.
 
+### Programming Paradigm
```
Comment (ErikSchierboom) [inline languages/csharp/reference/README.md]: I'm not entirely sure this adds much to the Concepts document to be honest. This is more like something for the C# track's language description. Could you perhaps remove this from the PR?

Comment (senal) [reply languages/csharp/reference/README.md]: Hi @ErikSchierboom,

Happy to remove this from PR if you see this not fitting here.

Comment (ErikSchierboom) [reply languages/csharp/reference/README.md]: @senal If you could do that, great!

**Pull request closed without merging**